src/server.rs

```diff
@@ -248,14 +251,25 @@ struct Cli {
 	cli_port: String,
 }
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    	let contents = fs::read_to_string("./example-config.toml")
-        	.expect("Something went wrong reading the file");
+fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+    let data_dir = util::get_default_data_dir();
 
-    	let config: Config = toml::from_str(&contents)
-        	.expect("Could not deserialize the config file");
+    // Initialize the logger
+    let log_writer = util::init_logger(&data_dir)?;
 
+    // Initialize the logger and log the log file path
+    log::info!("Logging initialized. Log file located at: {:?}", data_dir.join("debug.log"));
+
+    let contents = fs::read_to_string("../../example-config.toml")
+        .expect("Something went wrong reading the file");
+
+    let config: Config = toml::from_str(&contents)
+        .expect("Could not deserialize the config file");
```

Comment:
“Could not deserialize the config file content” as here we succeed on the the read of the file at least

Reply:
Updated in latest push